### PR TITLE
refactor: replace U256 with u64 in BLOCKHASH

### LINKED ISSUE #1423: Don't use U256 operation in EIP-2935 BLOCKHASH
since the window is lower than u64::MAX we can use the lowest word and use u64 modulo operation instead of U256 for better performance.

### PR DESCRIPTION
**Pull request by TropicalDog17**
## Description

Change the type of the number parameter in the block_hash methods of Host and Database from U256 to u64.

Closes #1423 

\cc @rakita @DaniPopes

### CODE REVIEW

File: crates/revm/src/context.rs [hunk 1]
```diff
@@ -108,9 +108,9 @@ impl<EXT, DB: Database> Host for Context<EXT, DB> {
         &mut self.evm.env
     }
 
-    fn block_hash(&mut self, number: U256) -> Option<B256> {
+    fn block_hash(&mut self, number: u64) -> Option<B256> {
         let block_number = as_usize_saturated!(self.env().block.number);
-        let requested_number = as_usize_saturated!(number);
+        let requested_number = number as usize;
```
Comment (rakita) [inline crates/revm/src/context.rs:113]: This can panic. We should use `usize::try_from(number).unwrap_or(usize::MAX);`

**Review by rakita — CHANGES_REQUESTED** [review]
cool change. One edge case pending

**Comment by TropicalDog17** [comment]
@rakita thanks for your review, I've refactored accordingly.

**Review by rakita — APPROVED** [review]
lgtm

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**